src/core/Core.js

```diff
@@ -415,7 +415,7 @@ class Uppy {
       throttledCalculateProgress(data)
     })
 
-    this.on('core:upload-success', (fileID, uploadResp, uploadURL) => {
```

Comment:
Found after re-reading:

>  I also removed the uploadResp parameter from the core:upload-success event, because it assumed that the response would be in JSON … It's a useful parameter though so we should find a way to add it back

Can’t we just return whatever the server returns, not promising it will be valid JSON? Is the fact that it needs to be handled through events the issue here?